ui/components/src/StatChart/StatChart.tsx

```diff
@@ -104,24 +104,43 @@ export function StatChart(props: StatChartProps) {
 
   const containerPadding = `${chartsTheme.container.padding.default}px`;
 
+  const textAlignment = sparkline ? 'auto' : 'center';
+  const textStyles = {
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: textAlignment,
+    alignItems: textAlignment,
+  };
+
   return (
-    <Box sx={{ height: '100%', width: '100%', display: 'flex', flexDirection: 'column' }}>
-      <Typography
-        variant="h3"
-        sx={(theme) => ({
-          color: color ?? theme.palette.text.primary,
-          fontSize: `clamp(${MIN_VALUE_SIZE}px, ${valueSize}px, ${MAX_VALUE_SIZE}px)`,
-          padding: `${containerPadding} ${containerPadding} 0 ${containerPadding}`,
-        })}
-      >
-        {formattedValue}
-      </Typography>
+    <Box sx={{ height: '100%', width: '100%', ...textStyles }}>
+      <Box sx={{ flexGrow: 1, ...textStyles }}>
+        <Typography
+          variant="h3"
+          sx={{
+            color,
+            fontSize: `clamp(${MIN_VALUE_SIZE}px, ${valueSize}px, ${MAX_VALUE_SIZE}px)`,
+            padding: `${containerPadding} ${containerPadding} 0 ${containerPadding}`,
+          }}
+        >
+          {formattedValue}
+        </Typography>
+      </Box>
       {sparkline !== undefined && (
-        <Box sx={{ flex: 1 }}>
+        <Box
```

Comment:
If the stat chart fills the entire content of the panel, isn't there a risk of the line and the text overlapping, making the text hard to read?

Reply:
+1 I'm curious about the wrapping behavior

Reply:
Sorry to add scope creep to this discussion, this is separate from this PR, but while I'm thinking about it--this [Node Exporter dashboard](https://grafana.com/grafana/dashboards/1860-node-exporter-full/), is one of the most ubiquitous Grafana dashboards out there. It has a ton of Stat charts all displayed together at really small dimensions (which really throws off our panel header too and Gauge chart), here is how it looks when using the Perses app migration feature: 

![image](https://user-images.githubusercontent.com/9369625/231624181-86c64ea6-eadf-4a5a-a41a-d4c3f33fc451.png)

Reply:
Yeah i notice grafana does a lot of auto formatting with stat charts. Here it automatically hides the sparkline when chart gets too small.  The problem also gets way more complicated once we support multiple series for stat charts

![stat chart](https://user-images.githubusercontent.com/9817826/232120391-86282953-c409-48c6-9627-19a4cac3eae2.gif)

Reply:
> If the stat chart fills the entire content of the panel, isn't there a risk of the line and the text overlapping, making the text hard to read?

Fixed